src/main/java/io/vlingo/symbio/store/object/jdbc/jpa/JPAObjectStoreActor.java

```diff
@@ -7,47 +7,139 @@
 package io.vlingo.symbio.store.object.jdbc.jpa;
 
 import io.vlingo.actors.Actor;
+import io.vlingo.actors.Definition;
+import io.vlingo.actors.Logger;
+import io.vlingo.common.Failure;
+import io.vlingo.common.Success;
+import io.vlingo.common.identity.IdentityGenerator;
+import io.vlingo.symbio.Entry;
+import io.vlingo.symbio.EntryAdapterProvider;
 import io.vlingo.symbio.Metadata;
 import io.vlingo.symbio.Source;
+import io.vlingo.symbio.State;
+import io.vlingo.symbio.StateAdapterProvider;
+import io.vlingo.symbio.store.Result;
+import io.vlingo.symbio.store.StorageException;
+import io.vlingo.symbio.store.dispatch.Dispatchable;
+import io.vlingo.symbio.store.dispatch.Dispatcher;
+import io.vlingo.symbio.store.dispatch.DispatcherControl;
+import io.vlingo.symbio.store.dispatch.control.DispatcherControlActor;
 import io.vlingo.symbio.store.object.PersistentObject;
 import io.vlingo.symbio.store.object.PersistentObjectMapper;
 import io.vlingo.symbio.store.object.QueryExpression;
 
+import java.time.LocalDateTime;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
 
 /**
  * JPAObjectStoreActor
  */
 public class JPAObjectStoreActor extends Actor implements JPAObjectStore {
+  private final DispatcherControl dispatcherControl;
+  private final Dispatcher<Dispatchable<Entry<String>, State<String>>> dispatcher;
   private boolean closed;
   private final JPAObjectStoreDelegate delegate;
+  private final EntryAdapterProvider entryAdapterProvider;
+  private final Logger logger;
+  private final IdentityGenerator identityGenerator;
+  private final StateAdapterProvider stateAdapterProvider;
 
-  public JPAObjectStoreActor(final JPAObjectStoreDelegate delegate) {
+  public JPAObjectStoreActor(final JPAObjectStoreDelegate delegate,
+          final Dispatcher<Dispatchable<Entry<String>, State<String>>> dispatcher) {
+    this(delegate, dispatcher, 1000L, 1000L);
+  }
+
+  public JPAObjectStoreActor(final JPAObjectStoreDelegate delegate,
+          final Dispatcher<Dispatchable<Entry<String>, State<String>>> dispatcher,
+          final long checkConfirmationExpirationInterval, final long confirmationExpiration) {
     this.delegate = delegate;
+    this.dispatcher = dispatcher;
     this.closed = false;
+    this.entryAdapterProvider = EntryAdapterProvider.instance(stage().world());
+    this.stateAdapterProvider = StateAdapterProvider.instance(stage().world());
+    this.logger = stage().world().defaultLogger();
+    this.identityGenerator = new IdentityGenerator.TimeBasedIdentityGenerator();
```

Comment:
Here and elsewhere, I don't know if the `TimeBasedIdentityGenerator` buys anything special. Is the internal timestamp actually being used? If not then a version 4 would be more efficient because this version 2 generator synchronizes threads in some/all cases. The default Java `UUID` is version 4.

Reply:
Changed to `IdentityGenerator.RandomIdentityGenerator()`